src/julia-syntax.scm

```diff
@@ -1950,6 +1950,34 @@
                ,@(map expand-forms (cddr e))))
         (cons (car e) (map expand-forms (cdr e))))))
 
+(define (expand-vcat e
+                     (vcat '((top vcat)))
+                     (hvcat '((top hvcat)))
+                     (hvcat_rows '((top hvcat_rows))))
+  (let ((a (cdr e)))
+    (if (any assignment? a)
+        (error (string "misplaced assignment statement in \"" (deparse e) "\"")))
+    (if (has-parameters? a)
+        (error "unexpected semicolon in array expression")
+        (expand-forms
+         (if (any (lambda (x)
+                    (and (pair? x) (eq? (car x) 'row)))
+                  a)
+             ;; convert nested hcat inside vcat to hvcat
+             (let ((rows (map (lambda (x)
+                                (if (and (pair? x) (eq? (car x) 'row))
+                                    (cdr x)
+                                    (list x)))
+                              a)))
+               ;; in case there is splatting inside `hvcat`, collect each row as a
+               ;; separate tuple and pass those to `hvcat_rows` instead (ref #38844)
+               (if (any (lambda (row) (any vararg? row)) rows)
+                   `(call ,.hvcat_rows ,.(map (lambda (x) `(tuple ,.x)) rows))
```

Comment:
Use `,@` everywhere instead of `,.`.

Reply:
Are the differences between the two explained anywhere? I always thought they were the same. I typically turn to the Racket manual for stuff like this, but they seem to only have `,@` for `unquote-splicing`.

Reply:
They do the same thing except `,.` is mutating, so it's a bit of an archaic micro-optimization.

Reply:
Ah, I see. So that's probably why it caused problems when dealing with multiple splicing interpolations.